.github/container/nsys_jax/nsys_jax/scripts/nsys_jax.py

```diff
@@ -0,0 +1,799 @@
+import shlex
+import shutil
+import sqlite3
+import subprocess
+import sys
+import tempfile
+import time
+import traceback
+import zipfile
+
+from .utils import execute_analysis_script, shuffle_analysis_arg
+from ..version import __sha__ as jax_toolbox_sha_with_prefix
+
+
+# Expand %q{ENV_VAR} if the variable is defined.
+def expand(string: str, skip_missing=True) -> str:
+    missing = set()
+
+    def rep(x):
+        if len(x.group(1)) % 2 == 0:
+            return x.group(0)
+        if x.group(2) not in os.environ:
+            missing.add(x.group(2))
+            return x.group(0)
+        return x.group(1)[:-1] + os.environ[x.group(2)]
+
+    expanded = re.sub(r"([%]+)q\{(.*?)\}", rep, string).replace("%%", "%")
+    if not skip_missing and missing:
+        raise Exception(f"{missing} not defined when expanding '{string}'")
+    return expanded
+
+
+# Use deflate compression
+COMPRESS_DEFLATE = {"compress_type": zipfile.ZIP_DEFLATED}
+# Do not compress (if the file is already compressed)
+COMPRESS_NONE: dict[str, int] = {}
+
+install_script_template = r"""#!/bin/bash
+#
+# Usage: ./install.sh [optional arguments to virtualenv]
+#
+# If it doesn't already exist, this creates a virtual environment named
+# `nsys_jax_env` in the current directory and installs Jupyter Lab and the
+# dependencies of the Analysis.ipynb notebook that is shipped alongside this
+# script inside the output archives of the `nsys-jax` wrapper.
+#
+# The expectation is that those archives will be copied and extracted on a
+# laptop or workstation, and this installation script will be run there, while
+# the `nsys-jax` wrapper is executed on a remote GPU cluster.
+set -ex
+SCRIPT_DIR=$(cd -- "$( dirname -- "${{BASH_SOURCE[0]}}" )" &> /dev/null && pwd)
+VIRTUALENV="${{SCRIPT_DIR}}/nsys_jax_venv"
+BIN="${{VIRTUALENV}}/bin"
+if [[ ! -d "${{VIRTUALENV}}" ]]; then
+  # Let `virtualenv` find/choose a Python. Currently >=3.10 is supported.
+  virtualenv -p 3.13 -p 3.12 -p 3.11 -p 3.10 "$@" "${{VIRTUALENV}}"
+  "${{BIN}}/pip" install -U pip
+  "${{BIN}}/pip" install 'nsys-jax[jupyter] @ git+https://github.com/NVIDIA/JAX-Toolbox.git@{jax_toolbox_commit}#subdirectory=.github/container/nsys_jax'
+  "${{BIN}}/install-flamegraph" "${{VIRTUALENV}}"
+  "${{BIN}}/install-protoc" "${{VIRTUALENV}}"
+else
+  echo "Virtual environment already exists, not installing anything..."
+fi
+if [ -z ${{NSYS_JAX_INSTALL_SKIP_LAUNCH+x}} ]; then
+  # Pick up the current profile data by default
+  export NSYS_JAX_DEFAULT_PREFIX="${{PWD}}"
+  # https://setuptools.pypa.io/en/latest/userguide/datafiles.html#accessing-data-files-at-runtime
+  NOTEBOOK=$("${{BIN}}/python" -c 'from importlib.resources import files; print(files("nsys_jax") / "analyses" / "Analysis.ipynb")')
+  echo "Launching: cd ${{SCRIPT_DIR}} && ${{BIN}}/jupyter-lab ${{NOTEBOOK}}"
+  cd "${{SCRIPT_DIR}}" && "${{BIN}}/jupyter-lab" "${{NOTEBOOK}}"
+else
+  echo "Skipping launch of Jupyter Lab due to NSYS_JAX_INSTALL_SKIP_LAUNCH"
+fi
+"""
```

Comment:
This replaces the old `install.sh`, otherwise the content of this file is mostly the old `nsys-jax`.